File: week1/java/week1_14.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] d = new int[n+1];
        d[1] = 1;
        d[2] = 2;
        for(int i=3; i<=n; i++){
            d[i] = (d[i-1] + d[i-2])%1000000007;
        }
        return d[n]%1000000007;
    }
}

File: week1/c++/week1_14.cpp
#include <string>
#include <vector>

using namespace std;

int d[60001];
long long int MOD = 1000000007;

int go(int i)
{
    if (d[i] > 0)
    {
        return d[i];
    }

    if (i <= 2)
    {
        return i;
    }

    d[i] = (go(i - 1) + go(i - 2)) % MOD;
    return d[i];
}

int solution(int n)
{

    int answer = go(n);
    return answer;
}